### PR TITLE
Converted to property expression

### PR DESCRIPTION
**Pull request by iBlitzkriegi**
Things like `something of %object%` are made easy with SimplePropertyExpression. It registers two syntaxes: `the formatted date of %player%` and `%player%'s formatted date`

### CODE REVIEW

**Comment by antoniojamore** [comment]
Pro skript add-on developer right here

File: src/me/wheezygold/skLib/skript/ExprFormatDateOfPlayer.java [hunk 1]
```diff
@@ -1,74 +1,53 @@
...
-import ch.njol.skript.lang.Expression;
-import ch.njol.skript.lang.ExpressionType;
-import ch.njol.skript.lang.SkriptParser.ParseResult;
-import ch.njol.skript.lang.util.SimpleExpression;
-import ch.njol.util.Kleenean;
-import me.wheezygold.skLib.common.JsonReader;
-
-public class ExprFormatDateOfPlayer extends SimpleExpression<String> {
-	
+public class ExprFormatDateOfPlayer extends SimplePropertyExpression<Object, String>{
```
Comment (Snow-Pyon) [inline src/me/wheezygold/skLib/skript/ExprFormatDateOfPlayer.java]: Why are you converting an Object instead of just convert Player?

**Review by Snow-Pyon — CHANGES_REQUESTED** [review]

File: src/me/wheezygold/skLib/skript/ExprFormatDateOfPlayer.java (hunk 1 above)
Comment (iBlitzkriegi) [reply src/me/wheezygold/skLib/skript/ExprFormatDateOfPlayer.java]: Woopsie

File: src/me/wheezygold/skLib/skript/ExprFormatDateOfPlayer.java [hunk 2]
```diff
@@ -1,74 +1,53 @@
...
+	protected String getPropertyName() {
+		return "formatted date";
 	}
 
-	@SuppressWarnings("unchecked")
 	@Override
-	public boolean init(Expression<?>[] arg0, int arg1, Kleenean arg2, ParseResult arg3) {
-		player = (Expression<Player>) arg0[0];
-		return true;
+	public String convert(Object o) {
```
Comment (Snow-Pyon) [inline src/me/wheezygold/skLib/skript/ExprFormatDateOfPlayer.java]: You forgot to change this line, line 30 and line 31.

File: src/me/wheezygold/skLib/skript/ExprFormatDateOfPlayer.java [hunk 3]
```diff
@@ -1,74 +1,53 @@
...
+		if(o instanceof Player){
+			InetAddress rawip = ((Player) o).getAddress().getAddress();
```
Comment (Snow-Pyon) [inline src/me/wheezygold/skLib/skript/ExprFormatDateOfPlayer.java]: Bad practice, not using camelCase in variable names. same in line 33 and 39.

**Review by Snow-Pyon — CHANGES_REQUESTED** [review]

File: src/me/wheezygold/skLib/skript/ExprFormatDateOfPlayer.java (hunk 3 above)
Comment (iBlitzkriegi) [reply src/me/wheezygold/skLib/skript/ExprFormatDateOfPlayer.java]: I just converted his code into a SimplePropertyExpression, didnt go through the code

**Comment by Snow-Pyon** [comment]
Looks good, other than what I pointed out on the reviews, I think this should be merged.

File: src/me/wheezygold/skLib/skript/ExprFormatDateOfPlayer.java (hunk 3 above)
Comment (Snow-Pyon) [reply src/me/wheezygold/skLib/skript/ExprFormatDateOfPlayer.java]: There should be no problem on changing that while you're doing this pull request though, if you don't mind.

**Comment by JRoy** [comment]
Fixed, fuck git

**Pull request closed without merging**